Sources/SystemInternals/Syscalls.swift

```diff
@@ -74,9 +85,27 @@ public func system_open(_ path: UnsafePointer<CChar>, _ oflag: Int32) -> CInt {
 #if ENABLE_MOCKING
   if mockingEnabled { return mock(String(cString: path), oflag) }
 #endif
+#if os(Windows)
+  var fh: CInt = -1
+  _ = _sopen_s(&fh, path, oflag, _SH_DENYNO, _S_IREAD | _S_IWRITE)
+  return fh
+#else
   return open(path, oflag)
+#endif
 }
 
+#if os(Windows)
+public func system_open(
+  _ path: UnsafePointer<CChar>, _ oflag: Int32, _ mode: CInt
+) -> CInt {
+#if ENABLE_MOCKING
+  if mockingEnabled { return mock(String(cString: path), oflag, mode) }
+#endif
+  var fh: CInt = -1
+  _ = _sopen_s(&fh, path, oflag, _SH_DENYNO, _S_IREAD | _S_IWRITE)
+  return fh
+}
```

Comment:
`mode` is not used, so why do we care to type it as a `CInt`? For Windows, would we not have a permissions parameter for `FileDescriptor.open`?.

Reply:
There is no `mode_t` type, and I don't think that you can use an internal typealias on a public func.

Reply:
Should we make the `FilePermissions` OptionSet empty on Windows? I'm a little uneasy about ignoring information.

Reply:
no, there is the concept of file permissions - its just that the permissions are different, there are no group permissions, and there are no execute permissions - only read/write.

Reply:
Ah, then we should we probably have group/execute unavailable on Windows, and the implementation of open here should do the read/write permission setting.